Assets/_Main/Scripts/UI/PlayerHealthUIManager.cs

```diff
@@ -0,0 +1,138 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using System;
+
+/// <summary>
+/// --Health UI Manager--<para></para>
+/// 
+/// This script controls the UI for the health of the player
+/// Also has the upper limit for the number of hearts
+/// Its subcribed to the player events and updates the UI accordingly
+/// </summary>
+/// <remarks> Execution order should be after PlayerHealth. </remarks>
+/// 
+public class PlayerHealthUIManager : MonoBehaviour
+{
+    #region FIELDS
+
+    [Header("Player Data")]
+    [Tooltip("Singleton PlayerHealth script")]
+    [SerializeField] private PlayerHealth _playerHealth;
+
+    [Header("Hearts")]
+    [Tooltip("Gameobject prefab with the image of the full heart")]
+    [SerializeField] private GameObject _heart;
+    [SerializeField] private Image _heartImage;
+    [Tooltip("Gameobject prefab with the image of the hollow heart")]
+    [SerializeField] private GameObject _hollowHeart;
+    [SerializeField] private Image _hollowHeartImage;
+
+    private struct HeartUI // "Ivan" Como se nombran los structs
+    {
+        public GameObject HeartGO;
+        public Image HeartIMG;
+        public Animator HeartAnim;
+    }
+
+    private class HeartDictionary : Dictionary<int, HeartUI> // "Ivan" Como se nombran los diccionaries
+    {
+        private HeartUI newHeart;
+        internal void Add(int key, GameObject heartGO)
```

Comment:
You can make this public, internal means that is contained only on the assembly, but Unity by default only uses one assembly

Reply:
Interesting implementation, may be an overkill for just some hearts but is interesting.